hyrisecockpit/drivers/__default__/driver.py

```diff
@@ -0,0 +1,133 @@
+"""Module for default workload behavior."""
+
+from time import time_ns
+from typing import Dict, List, Optional, Tuple, Union
+
+from psycopg2.extensions import AsIs
+
+from hyrisecockpit.drivers.__default__.default_workload import DefaultWorkload
+
+
+class DefaultDriver:
+    """Driver for default behavior."""
+
+    def __init__(
+        self,
+        query_path: str,
+        scale_factor_query_path: Dict,
+        benchmark_type: str,
+        table_names: List[str],
+    ):
+        """Initialize DefaultDriver."""
+        self._query_path: str = query_path
+        self._scale_factor_query_path: Dict = scale_factor_query_path
+        self._benchmark_type: str = benchmark_type
+        self._workloads: Dict = {}
+        self._table_names = table_names
+
+    def _get_workload_for_scale_factor(self, scalefactor):
+        workload = self._workloads.get(scalefactor)
+        if workload is None:
+            query_path = (
+                f"{self._query_path}/{self._scale_factor_query_path[scalefactor]}"
+            )
+            workload = DefaultWorkload(self._benchmark_type, scalefactor, query_path)
+            self._workloads[scalefactor] = workload
+
+        return workload
+
+    def generate(self, scalefactor, frequency, weights):
+        """Generate workload queries."""
+        workload = self._get_workload_for_scale_factor(scalefactor)
+        return workload.get(frequency, weights)
+
+    def _get_formatted_scalefactor(self, scalefactor: int) -> str:
+        """Get formatted scalefactor."""
+        if scalefactor == int(scalefactor):
+            formatted_scalefactor = str(int(scalefactor))
+        else:
+            formatted_scalefactor = str(scalefactor).replace(".", "_")
+
+        return formatted_scalefactor
```

Comment:
The type hint for `scalefactor` does not make sense for the conditional expression.
```suggestion
    def _get_formatted_scalefactor(self, scalefactor: Union[int, float]) -> str:
        """Get formatted scalefactor."""
        if scalefactor == int(scalefactor):
            scalefactor = int(scalefactor)
        return str(scalefactor).replace(".", "_")
```